### DISCUSSION TITLE
Question about truststore and expired certs

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#4705] [Q&A]
Can anyone comment on what needs to be done to stop an expired cert from showing a warning? I assumed the support was removing the certs from the truststore but that doesn't seem to be the case, as I can see expired certs in there but they do not throw a warning. In the image below, the identrustdstx3 is about to expire and is giving a warning when logging into mirth. Can it just be removed from the truststore or is there another way to do this?

![image](https://user-images.githubusercontent.com/6484700/131847485-c47aac59-86dc-451f-afad-cc3830be843a.png)

**Comment by ab-20-m** [comment]
Check the cacerts file of the JRE/JDK the Mirth Server/Service is running from. I don't believe Mirth has used the truststore.jks since 3.0.X but my memory is fuzzy.

**Comment by rutledj** [comment]
Thanks. Looked in all the folders in the appliance but not seeing a cacerts file.

**Reply by JeffBenedict** [reply]
did you ever get an answer? seems MC running on appliances act differently than on other installs.

**Reply by jonbartels** [reply]
The answer for both may be that the underlying JDK or OS needs updated to update the cacerts provided by the JVM.

A major part of Java patch releases (eg 1.8_123 -> 1.8_456) would be updates to the JVM cacerts to update roots. 
Most Linux distros provide a "java-cacerts" package which ONLY updates cacerts without the other Java patch updates.

For appliance users - You're paying for an appliance and the associated annual fees. Ask NextGen. Under the covers its just a CentOS distro with a good management web UI.
